dev-docs/007-transforms-rfc.md

```diff
@@ -0,0 +1,99 @@
+- Start Date: 2019-09-05
+- RFC PR: 
+
+# Summary
+
+It is desirable to enable arbitrary transform support common in robotic systems. While XVIZ models some explicitly
+with our coordinate types IDENTITY, GEOGRAPHIC, and VEHICLE_RELATIVE, the need for client controlled frames of reference
+will make it easier to interop with other systems as well as match a familiar modeling automous system data.
+
+# Motivation
+
+Some PERSISTENT data, such as *maps* and *point clouds*, could be encoded in various ways such as vehicle relative.
+If they are semantically fixed, but the reference point changes, there is no way to efficiently update them without
+sending all the points again.
+
+Ideally, the only data that is required is the transform to adjust how the points are transformed.
+
+Extending this to a general robotic systems convention, the need to describe parent/child relationships
+is a common model for robotic systems.
+
+XVIZ is not focused on articulated robotic platforms but support for the convention common when modeling autonomous data
+would be benefitial for XVIZ and it's users.
+
+# Proposal
+
+### POSE schema
+ - Add a *parent* field to the pose that is a string that my reference another pose stream
```

Comment:
How does a static stream reference a pose more precisely? It does not have a pose itself right? Does it include the name of a pose stream?

Perhaps this is just a question of naming, I know that "poses" make a lot of sense for robotics, but I wonder if instead of `parent`, letting e.g. map streams name a `COORDINATE_SYSTEM` to reference another pose stream would be more natural/understandable to users?

Those static streams would snapshot an inverse transform from the coordinate system stream and then keep multiplying it with the latest transform from that stream?

Allowing the xviz log to introduce pose streams naming a couple of "coordinate systems" and keep updating their relative poses. But I guess that is what you are proposing.

Reply:
> How does a static stream reference a pose more precisely? It does not have a pose itself right? Does it include the name of a pose stream?

from some comments from Joe, I believe we will enable stream metadata to be sent at anytime, so we can set the stream property for which pose it references there by name.

For the naming, I used *parent* in the context of the pose as in a scene graph that would be pretty natural.  *COORDINATE_SYSTEM* or even *FRAME* are also common terms.  I'm not tied to anything specific here.

> Allowing the xviz log to introduce pose streams naming a couple of "coordinate systems" and keep updating their relative poses. But I guess that is what you are proposing.

precisely

Reply:
In discussion with Joe we have settled on referring to the connections as source -> target.  And since we define transforms with our pose streams, we will use the name "source_pose" in our link structure.

The connection to a primitive stream and a pose will be created not in either existing type, but in a new *links* mapping that will connect the target to the source.  Details in an updated RFC will land shortly.